apps/microsoft-teams/frontend/src/components/config/ConfigPage/ConfigPage.spec.tsx

```diff
@@ -1,29 +1,82 @@
 import ConfigPage from './ConfigPage';
 import { describe, expect, it, vi } from 'vitest';
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import { mockSdk } from '@test/mocks';
-import { headerSection } from '@constants/configCopy';
+import { headerSection, notificationsSection } from '@constants/configCopy';
 
 vi.mock('@contentful/react-apps-toolkit', () => ({
   useSDK: () => mockSdk,
 }));
 
+vi.mock('@azure/msal-react', () => ({
+  useMsal: vi.fn().mockReturnValue({ accounts: [{ asdf: 'asdf' }] }),
+}));
+
 vi.mock('@components/config/AccessSection/AccessSection', () => ({
   default: () => {
     return <div>Mock Access Section</div>;
   },
 }));
 
 vi.mock('@components/config/NotificationsSection/NotificationsSection', () => ({
-  default: () => {
-    return <div>Mock Notification Section</div>;
+  default: ({
+    setNotificationIndexToEdit,
+  }: {
+    setNotificationIndexToEdit: (idx: number) => void;
+  }) => {
+    return (
+      <>
+        <h1>MOCKED NOTIFICATIONS!!!!!</h1>
+        <button
+          onClick={() => {
+            setNotificationIndexToEdit(0);
+          }}
+          data-testid="edit-notification-btn">
+          edit
+        </button>
+      </>
+    );
   },
 }));
 
 describe('ConfigPage component', () => {
   it('mounts and renders title', () => {
-    render(<ConfigPage />);
+    const { unmount } = render(<ConfigPage />);
 
     expect(screen.getByText(headerSection.title)).toBeTruthy();
+
+    unmount();
+  });
+
+  describe('Saving via WidgetRenderer', () => {
+    it('prompts the User to save or cancel their pending changes when they have an open Notification modal', async () => {
+      const { unmount } = render(<ConfigPage />);
+
+      // Open an existing notification to edit it, i.e. pending change
+      let editButton;
```

Comment:
i am not a vitest expert but per https://vitest.dev/api/vi#vi-waitfor-0-34-5 couldn't we restructure this a bit to just return the edit button from the function?

```
const editButton = await waitFor(() => {
        // waitFor async callbacks to resolve inside useEffects
        return screen.getByTestId('edit-notification-btn');
      });
      
  if (editButton) // ...
```

Reply:
+1 to this solution. Keeping everything `const` is always nice

Reply:
Nice, this is much better!  

- [ ] implement this.